ostelco-core/Network/PrimeAPI.swift

```diff
@@ -118,7 +119,11 @@ open class PrimeAPI: BasicNetwork {
     /// - Parameter userSetup: The `UserSetup` to use.
     /// - Returns: A promise which when fullfilled will contain the created customer model.
     public func createCustomer(with userSetup: UserSetup) -> Promise<CustomerModel> {
-        return self.sendObject(userSetup, to: RootEndpoint.customer.value, method: .POST)
+        let queryItems = [
+            URLQueryItem(name: "nickname", value: userSetup.nickname),
+            URLQueryItem(name: "contactEmail", value: userSetup.contactEmail)
```

Comment:
Ugh, missed that `withParam` was query parameters even for `POST` requests. Sorry about that. 

...also why are we using query parameters instead of JSON for `POST` requests?